broker/src/main/java/de/fraunhofer/ids/messaging/broker/BrokerService.java

```diff
@@ -190,24 +210,28 @@ public MessageProcessedNotificationMAP updateSelfDescriptionAtBroker(@NonNull fi
      * @param brokerURIs
      */
     @Override
-    public List<MessageProcessedNotificationMAP> updateSelfDescriptionAtBrokers(@NonNull final List<URI> brokerURIs) {
+    public List<MessageProcessedNotificationMAP> updateSelfDescriptionAtBrokers(@NonNull final List<URI> brokerURIs)
+            throws
+            IOException,
+            DeserializeException,
+            ShaclValidatorException,
+            UnexpectedResponseException,
+            SerializeException,
+            DapsTokenManagerException,
+            MultipartParseException,
+            ClaimsException,
+            SendMessageException,
+            UnknownResponseException,
+            MessageBuilderException {
         final ArrayList<MessageProcessedNotificationMAP> responses = new ArrayList<>();
 
         for (final var uri : brokerURIs) {
-            try {
-                final var response = updateSelfDescriptionAtBroker(uri);
-
-                if (log.isInfoEnabled()) {
-                    log.info(String.format("Received response from %s", uri));
-                }
-                responses.add(response);
-
-            } catch (IOException | MultipartParseException | ClaimsException | DapsTokenManagerException | UnknownResponseException | DeserializeException | UnexpectedResponseException e) {
-                if (log.isWarnEnabled()) {
-                    log.warn(String.format("Connection to Broker %s failed!", uri));
-                    log.warn(e.getMessage(), e);
-                }
```

Comment:
Information lost?

Reply:
Strictly speaking for the connector developer rather more information than less, because the exception are not all caught and just logged. So the exceptions are passed on to the connector developer, who can then act.